### PR TITLE
Relax Assertion in MultiFileWriter

### PR DESCRIPTION
**Pull request by original-brownbear**
As discussed, we want to relax this.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public void writeFile(StoreFileMetadata fileMetadata, long readSnapshotFileBuffe
             }
 
             Store.verify(indexOutput);
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java:120]: Can we also only do this when `verifyOutput` is `true`?

File: server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java [hunk 2]
```diff
@@ -118,7 +118,7 @@ public void writeFile(StoreFileMetadata fileMetadata, long readSnapshotFileBuffe
...
-            assert Arrays.asList(store.directory().listAll()).contains(tempFileName)
+            assert verifyOutput == false || Arrays.asList(store.directory().listAll()).contains(tempFileName)
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java:121]: :+1:

I suspect we need something similar in `innerWriteFileChunk` too?

**Review by tlrx — COMMENTED** [review]
Looks good, I left some comments

File: server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java (hunk 2 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java:121]: I think we're not writing chunks for where we're looking to use this are we?

File: server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java]: ++ done

File: server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java (hunk 2 above)
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java]: I double checked and you're right, it should not be needed

**Review by tlrx — APPROVED** [review]
LGTM assuming CI is happy

**Comment by original-brownbear** [comment]
Thanks Tanguy!

**Pull request merged**